backend/src/v2/component/launcher_v2.go

```diff
@@ -562,51 +562,53 @@ func downloadArtifacts(ctx context.Context, executorInput *pipelinespec.Executor
 		if len(artifactList.Artifacts) == 0 {
 			continue
 		}
-		inputArtifact := artifactList.Artifacts[0]
```

Comment:
There are other parts of the code assume the artifacts list is either a size of 0 or 1. See `addModelcarsToPodSpec` as an example. So at the very least, `addModelcarsToPodSpec` probably needs to be adjusted to look at the whole list as well or else `waitForModelcar` may hang forever for a model car that was never added as a side car container.